File: backend/src/main/java/com/mini/credit/dto/caisse/DepenseCaisseRattachementTransportRequest.java
package com.mini.credit.dto.caisse;

import com.mini.credit.enums.TypeChargeFixe;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class DepenseCaisseRattachementTransportRequest {
    @NotNull
    private Long employeId;
    @NotBlank
    private String periodeCharge;
    private LocalDate dateDebutPeriode;
    private LocalDate dateFinPeriode;
    @NotNull
    private Long siteId;
    @NotNull
    private TypeChargeFixe typeChargeFixe;
    @NotBlank
    private String commentaireCorrection;
}

File: frontend/src/app/features/caisse/models/depense-caisse-rattachement-transport-request.ts
export interface DepenseCaisseRattachementTransportRequest {
  employeId: number;
  periodeCharge: string;
  dateDebutPeriode?: string;
  dateFinPeriode?: string;
  siteId: number;
  typeChargeFixe: 'TRANSPORT_SITE';
  commentaireCorrection: string;
}
